PWGHF/TableProducer/trackIndexSkimCreator.cxx

```diff
@@ -1452,667 +1486,748 @@ struct HfTrackIndexSkimCreator {
+      int nCutStatus3ProngBit = BIT(nCuts3Prong) - 1; // bit value for selection status for each 3-prong candidate where each selection is one bit and they are all set to 1
+
+      int whichHypo2Prong[n2ProngDecays];
+      int whichHypo3Prong[n3ProngDecays];
+
+      // set the magnetic field from CCDB
+      auto bc = collision.bc_as<o2::aod::BCsWithTimestamps>();
+      initCCDB(bc, runNumber, ccdb, isRun2 ? ccdbPathGrp : ccdbPathGrpMag, lut, isRun2);
+
+      // 2-prong vertex fitter
+      o2::vertexing::DCAFitterN<2> df2;
+      df2.setBz(o2::base::Propagator::Instance()->getNominalBz());
+      df2.setPropagateToPCA(propagateToPCA);
+      df2.setMaxR(maxR);
+      df2.setMaxDZIni(maxDZIni);
+      df2.setMinParamChange(minParamChange);
+      df2.setMinRelChi2Change(minRelChi2Change);
+      df2.setUseAbsDCA(useAbsDCA);
+
+      // 3-prong vertex fitter
+      o2::vertexing::DCAFitterN<3> df3;
+      df3.setBz(o2::base::Propagator::Instance()->getNominalBz());
+      df3.setPropagateToPCA(propagateToPCA);
+      df3.setMaxR(maxR);
+      df3.setMaxDZIni(maxDZIni);
+      df3.setMinParamChange(minParamChange);
+      df3.setMinRelChi2Change(minRelChi2Change);
+      df3.setUseAbsDCA(useAbsDCA);
+
+      // used to calculate number of candidiates per event
+      auto nCand2 = rowTrackIndexProng2.lastIndex();
+      auto nCand3 = rowTrackIndexProng3.lastIndex();
+
+      // if there isn't at least a positive and a negative track, continue immediately
+      // if (tracksPos.size() < 1 || tracksNeg.size() < 1) {
+      //  return;
+      //}
+
+      // first loop over positive tracks
+      // for (auto trackPos1 = tracksPos.begin(); trackPos1 != tracksPos.end(); ++trackPos1) {
+
+      auto groupedTrackIndices = trackIndices.sliceBy(trackIndicesPerCollision, collision.globalIndex());
+
+      for (auto trackIndexPos1 = groupedTrackIndices.begin(); trackIndexPos1 != groupedTrackIndices.end(); ++trackIndexPos1) {
+        auto trackPos1 = trackIndexPos1.track_as<SelectedTracks>();
+
+        if (trackPos1.signed1Pt() < 0) {
+          continue;
+        }
 
-    bool cutStatus2Prong[n2ProngDecays][nCuts2Prong];
-    bool cutStatus3Prong[n3ProngDecays][nCuts3Prong];
-    int nCutStatus2ProngBit = BIT(nCuts2Prong) - 1; // bit value for selection status for each 2-prong candidate where each selection is one bit and they are all set to 1
-    int nCutStatus3ProngBit = BIT(nCuts3Prong) - 1; // bit value for selection status for each 3-prong candidate where each selection is one bit and they are all set to 1
+        // // retrieve the selection flag that corresponds to this collision
+        int isSelProngPos1 = trackPos1.isSelProng();
+        for (auto iCollId{0u}; iCollId < trackPos1.compatibleCollIds().size(); ++iCollId) {
+          if (trackPos1.compatibleCollIds()[iCollId] == collision.globalIndex()) {
+            isSelProngPos1 = trackPos1.isSelProngAllColls()[iCollId];
+            break;
```

Comment:
I think it is fine and cannot be done faster for such a short list
(we would need to keep an iterator per track if we want to cash the previous position (the collisions are ordered)). I don't think that is worth it.

Reply:
Maybe we can make a lambda, to avoid the copy-paste with the inner track loop

Reply:
Why not use `std::find_if`?